tools/schemacode/schemacode/validator.py

```diff
@@ -658,6 +675,77 @@ def log_errors(validation_result):
         lgr.warning("The `%s` file was not matched by any regex schema entry.", i)
 
 
+def _inheritance_expansion(
+    regex_string,
+    datatype=None,
+    debug=False,
+):
+    """
+    Generate regex strings applying BIDS inheritance expansion to an input string.
+
+    Parameters
+    ----------
+    regex_string : str
+        String representing the regex to which inheritance expansion should be applied.
+    datatype : str, optional
+        Datatype string to remove as part of inheritance expansion.
+    debug : bool, optional
+        Whether to print which expansions were generated.
+
+    Returns
+    -------
+    expanded_regexes : list of str
+    """
+
+    # Order is important as the string is eroded.
+    # Session is eroded *together with* and *after* subject, as it is always optional
+    # and the erosion is:
+    #   * only required if a dangling leading underscore is present after subject removal.
+    #   * only BIDS-valid after the subject field is eroded from the filename.
+    expansions = [
+        {
+            "regex": [
+                r".*?(?P<remove>sub-\(\?P<subject>\(\[0\-9a\-zA\-Z\]\+\)\)/).*?",
+                r".*?(?P<remove>sub-\(\?P=subject\))",
+                r".*?/(?P<remove>\(\|ses-\(\?P<session>\(\[0\-9a\-zA\-Z\]\+\)\)/\)\(\|_ses-\("
+                r"\?P=session\)\)_).*?",
+            ],
+            "replace": ["", "", ""],
+        },
+    ]
```

Comment:
so it is only subject and session entities which are "eroded" or some other entities as well?   E.g. example in https://bids-specification.readthedocs.io/en/latest/02-common-principles.html#the-inheritance-principle would erode/remove `_acq-` from the sidecar file to consider top level task .json file to inherit.

Reply:
`_acq-` is already optional, so doesn't need to be eroded explicitly.